src/main/java/org/gridsuite/directory/server/DirectoryService.java

```diff
@@ -248,13 +248,19 @@ public List<ElementAttributes> getDirectoryElements(UUID directoryUuid, String u
     }
 
     private Stream<ElementAttributes> getDirectoryElementsStream(UUID directoryUuid, String userId, List<String> types) {
-        return getAllDirectoryElementsStream(directoryUuid, types)
+        return getAllDirectoryElementsStream(directoryUuid, types, userId)
                 .filter(elementAttributes -> !elementAttributes.getType().equals(DIRECTORY) || elementAttributes.isAllowed(userId));
     }
 
-    private Stream<ElementAttributes> getAllDirectoryElementsStream(UUID directoryUuid, List<String> types) {
+    private Stream<ElementAttributes> getAllDirectoryElementsStream(UUID directoryUuid, List<String> types, String userId) {
         List<DirectoryElementEntity> directoryElements = directoryElementRepository.findAllByParentId(directoryUuid);
-        Map<UUID, Long> subdirectoriesCountsMap = getSubElementsCount(directoryElements.stream().map(DirectoryElementEntity::getId).collect(Collectors.toList()), types);
+        Map<UUID, Long> subdirectoriesCountsMap;
+        if (!types.isEmpty()) {
+            subdirectoriesCountsMap = getSubElementsCount(directoryElements.stream().map(DirectoryElementEntity::getId).collect(Collectors.toList()), types);
```

Comment:
why don't we pass the user when the type is empty ?

Reply:
to keep the old method for special research and don't show the arrows for nothing, It's the same than the root